### PR TITLE
Sync Committee verification metrics

### PR DESCRIPTION
**Pull request by Bez625**
- Add Sync Committee Verification metrics to highlight when we were and were not included in a SyncAggregate. 
- Add config setting to enable the above metrics.

### CODE REVIEW

**Review by mcdee — CHANGES_REQUESTED** [review]
Looks good, a few tweaks to docs and associated commentary..

File: docs/metrics/prometheus.md [hunk 1]
```diff
@@ -138,3 +138,13 @@ There is also a companion metric `vouch_relay_auction_block_duration_seconds_cou
...
 
 `vouch_relay_validator_registrations_duration_seconds_bucket` is provided as a histogram, with buckets in increments of 0.1 seconds up to 4 seconds.  It provides details of the total time taken for Vouch to serve validator registration requests from beacon nodes.  There is also a companion metric `vouch_relay_validator_registrations_duration_seconds_count`, which is a simple count of the number of operations that have taken place.
+
+## Sync Committee Verification
+
+Sync Committee Verification metrics can be enabled using the `controller.verify-sync-committee-inclusion` flag in the configuration. This gives more insight in to the participation of Sync Committee duties:
+
+- `vouch_synccommitteeverification_current_assigned` is a gauge that is set to the current number of vouch validators that are participating in Sync Committee duty.
+- `vouch_synccommitteeverification_mismatches_total` is a counter that increments each time vouch receives a head event where the parent block root does not match the root vouch broadcast in the Sync Committee messages.
```
[the remark this answers is not in the document]
Comment (Bez625) [reply docs/metrics/prometheus.md:147]: Presently this is slot level, but I did consider if it should be validator level. Happy to make the change, but just to be clear if `mismatches+found+missing == len(validators)` then missing metrics would not include mismatch (although you would expect mismatch not to be present in SyncAggregate). 

I'll make this clear in the description of the metrics that `missing` would not include `mismatch` validators.

[the remark this answers is not in the document]
Comment (mcdee) [reply docs/metrics/prometheus.md:147]: I think validator level makes more sense; thanks.

File: services/controller/standard/events.go [hunk 2]
```diff
@@ -332,3 +343,85 @@ func (s *Service) subscribeToBeaconCommittees(ctx context.Context,
...
+		log.Debug().Err(err).Msg("Failed to get sync aggregate retrieved from head block")
+		return
+	}
+
+	// The previousSlotData.ValidatorToCommitteeIndex maps all of our validators to their committee indices for every slot in the epoch.
+	// Assumption here is that for every slot each validator should have contributed to all their committee indices for the whole epoch.
+	for validatorIndex, committeeIndices := range previousSlotData.ValidatorToCommitteeIndex {
+		for _, committeeIndex := range committeeIndices {
+			if !syncAggregate.SyncCommitteeBits.BitAt(uint64(committeeIndex)) {
+				log.Debug().Uint64("validator_index", uint64(validatorIndex)).
```
[the remark this answers is not in the document]
Comment (Bez625) [reply services/controller/standard/events.go:418]: Agreed. I'll make a note to do this as a subsequent task as I'd really like to get this merged.

**Review by mcdee — APPROVED** [review]

**Pull request merged**